rotkehlchen/exchanges/htx.py

```diff
@@ -188,24 +221,234 @@ def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
+
+        This logic processes all entries returned by the API, discarding those that do not fall
+        within the range start_ts <= timestamp <= end_ts. Additionally, it does not terminate the
+        querying process until it encounters an created_at value that is lower than the start_ts.
+        """
+        log.debug(f'querying HTX deposits/withdrawals with {start_ts=}-{end_ts=}')
+        try:
+            raw_data = self._paginated_query(
+                endpoint='/v1/query/deposit-withdraw',
+                options={
+                    'type': query_for,
+                    'size': PAGINATION_LIMIT,
+                    'direct': 'next',  # sorted by descending order
+                },
+                start_ts=start_ts,
+            )
+        except RemoteError as e:
+            log.error(f'Failed to query HTX api for deposits and withdrawals due to {e!s}')
+            return []
+
+        movements = []
+        for movement in raw_data:
+            if (timestamp_raw := movement.get('created-at')) is not None:
+                timestamp = ts_ms_to_sec(TimestampMS(int(timestamp_raw)))
+                if not start_ts <= timestamp <= end_ts:
+                    continue  # skip if is not in range, could come from last page
+            else:
+                log.error(f'Could not find timestamp for HTX movement {movement}. Skipping...')
+                continue
+
+            try:
+                coin = asset_from_htx(movement['currency'])
+                movements.append(AssetMovement(
+                    timestamp=timestamp,
+                    location=Location.HTX,
+                    category=AssetMovementCategory.DEPOSIT if query_for == 'deposit' else AssetMovementCategory.WITHDRAWAL,  # noqa: E501
+                    address=movement['address'],
+                    transaction_id=movement['tx-hash'],
+                    asset=coin,
+                    amount=deserialize_asset_amount(movement['amount']),
+                    fee_asset=coin,
+                    fee=deserialize_fee(movement.get('fee', '0')),
+                    link=str(movement['id']),
+                ))
+            except (DeserializationError, KeyError) as e:
+                msg = str(e)
+                if isinstance(e, KeyError):
+                    msg = f'Missing key {msg}'
+
+                log.error(f'{e} when reading movement for HTX deposit {movement}. Skipping...')
+                continue
+            except UnknownAsset:
+                log.error(f'Could not read assets from trade pair {movement["currency"]}. Skipping...')  # noqa: E501
+                continue
+
+        return movements
+
     def query_online_deposits_withdrawals(
             self,
             start_ts: Timestamp,
             end_ts: Timestamp,
     ) -> list[AssetMovement]:
         """Query deposits and withdrawals sequentially"""
-        return []  # TODO: Implement this in another PR
+        new_movements = self._query_deposits_withdrawals(
```

Comment:
https://github.com/rotki/rotki/pull/8135#discussion_r1649345796

You resolved this. But how? You did not write anything. Why did you not just do one call for both deposits and withdrawals? Why do you need a different call to the exchange for deposits and a different for withdrawals?

Reply:
I had mentioned it in this [comment](https://github.com/rotki/rotki/pull/8135#discussion_r1650573379) where you had asked the question. The API needs a `type` to get the deposits or withdrawals.